src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Devices/DataBoxEdgeDeviceInvokeCmdletBase.cs

```diff
@@ -0,0 +1,215 @@
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.EdgeGateway;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using PSResourceModel = Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Models.PSDataBoxEdgeDevice;
+
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Common.Cmdlets.Devices
+{
+    [Cmdlet(VerbsLifecycle.Invoke, Constants.Device, DefaultParameterSetName = InvokeScanForUpdateParameterSet,
+         SupportsShouldProcess = true
+     ),
+     OutputType(typeof(bool)),
+    ]
+    public class DataBoxEdgeDeviceInvokeCmdletBase : AzureDataBoxEdgeCmdletBase
+    {
+        private const string InvokeScanForUpdateParameterSet = "InvokeScanForUpdateParameterSet";
+
+        private const string InvokeScanForUpdateByResourceIdParameterSet =
+            "InvokeScanForUpdateByResourceIdParameterSet";
+
+        private const string InvokeScanForUpdateByDeviceObjectParameterSet =
+            "InvokeScanForUpdateByDeviceObjectParameterSet";
+
+        private const string InvokeFetchUpdateParameterSet = "InvokeFetchUpdateParameterSet";
+        private const string InvokeFetchUpdatesByResourceIdParameterSet = "InvokeFetchUpdatesByResourceIdParameterSet";
+
+        private const string InvokeFetchUpdatesByDeviceObjectParameterSet =
+            "InvokeFetchUpdatesByDeviceObjectParameterSet";
+
+        private const string InvokeInstallUpdateParameterSet = "InvokeInstallUpdateParameterSet";
+
+        private const string InvokeInstallUpdatesByResourceIdParameterSet =
+            "InvokeInstallUpdatesByResourceIdParameterSet";
+
+        private const string InvokeInstallUpdatesByDeviceObjectParameterSet =
+            "InvokeInstallUpdatesByDeviceObjectParameterSet";
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = InvokeFetchUpdatesByResourceIdParameterSet,
+            HelpMessage = Constants.ResourceIdHelpMessage)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = InvokeInstallUpdatesByResourceIdParameterSet,
+            HelpMessage = Constants.ResourceIdHelpMessage)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = InvokeScanForUpdateByResourceIdParameterSet,
+            HelpMessage = Constants.ResourceIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = true,
```

Comment:
Conflict with the design issue at https://github.com/raviten/azure-powershell/blob/raviten/databoxedge-cmdlets-ii/src/DataBoxEdge/DataBoxEdge/help/Invoke-AzDataBoxEdgeDevice.md#invokescanforupdateparameterset-default